packages/next/src/server/base-server.ts

```diff
@@ -816,6 +824,31 @@ export default abstract class Server<
     return matchers
   }
 
+  protected async instrumentationOnRequestError(
+    ...args: Parameters<ServerOnInstrumentationRequestError>
+  ) {
+    const [err, req, ctx] = args
+
+    if (
+      process.env.__NEXT_EXPERIMENTAL_INSTRUMENTATION &&
+      this.instrumentation
+    ) {
+      this.instrumentation.onRequestError?.(
```

Comment:
```suggestion
      await this.instrumentation.onRequestError?.(
```
Should this be awaited to ensure request isn't closed before it's finished?